GDJS/GDJS/Events/CodeGeneration/EventsCodeGenerator.cpp

```diff
@@ -1298,7 +1307,34 @@ gd::String EventsCodeGenerator::GenerateGetVariable(
     const gd::String& objectName) {
   gd::String output;
   const gd::VariablesContainer* variables = NULL;
-  if (scope == LAYOUT_VARIABLE) {
+  if (scope == ANY_VARIABLE) {
+    if (HasProjectAndLayout()) {
+      const auto variablesContainersList =
+          GetProjectScopedContainers().GetVariablesContainersList();
+      const auto &variablesContainer =
+          variablesContainersList.GetVariablesContainerFromVariableName(
+              variableName);
+      const auto sourceType = variablesContainer.GetSourceType();
+      if (sourceType == gd::VariablesContainer::SourceType::Scene) {
+        variables = &variablesContainer;
+        output = "runtimeScene.getScene().getVariables()";
+      } else if (sourceType == gd::VariablesContainer::SourceType::Global) {
+        variables = &variablesContainer;
+        output = "runtimeScene.getGame().getVariables()";
+      } else if (sourceType == gd::VariablesContainer::SourceType::Local) {
+        variables = &variablesContainer;
+        std::size_t localVariablesIndex =
+            variablesContainersList.GetLocalVariablesContainerPosition(
+                variablesContainer);
+        output = GetCodeNamespace() + ".localVariables[" +
+                 gd::String::From(localVariablesIndex) + "]";
+      }
+    }
+    else {
+      output = "// Unsupported unified variables in functions.";
+      output += "runtimeScene.getScene().getVariables()";
```

Comment:
Might be even safer to make an invalid code so that we don't silently fail or forget a case.